torch/csrc/generic/serialization.cpp

```diff
@@ -29,30 +29,35 @@ THTensor * THPTensor_(newWithMetadataFileRaw)(int fd, THStorage *storage)
 void THPStorage_(writeFileRaw)(THStorage *self, int fd)
 {
   real *data;
+  int64_t size = self->size;
 #ifndef THC_GENERIC_FILE
   data = self->data;
 #else
-  std::unique_ptr<char[]> cpu_data(new char[self->size * sizeof(real)]);
+  std::unique_ptr<char[]> cpu_data(new char[size * sizeof(real)]);
   data = (real*)cpu_data.get();
-  THCudaCheck(cudaMemcpy(data, self->data, self->size * sizeof(real), cudaMemcpyDeviceToHost));
+  THCudaCheck(cudaMemcpy(data, self->data, size * sizeof(real), cudaMemcpyDeviceToHost));
 #endif
-  SYSCHECK(write(fd, &self->size, sizeof(long)));
+  ssize_t result = write(fd, &size, sizeof(int64_t));
+  if (result != sizeof(int64_t))
+    throw std::system_error(result, std::system_category());
   // fast track for bytes and little endian
   if (sizeof(real) == 1 || THP_nativeByteOrder() == THPByteOrder::THP_LITTLE_ENDIAN) {
     char *bytes = (char *) data;
-    uint64_t remaining = sizeof(real) * self->size;
+    int64_t remaining = sizeof(real) * size;
     while (remaining > 0) {
       ssize_t result = write(fd, bytes, remaining);
       if (result < 0)
         throw std::system_error(result, std::system_category());
       bytes += result;
       remaining -= result;
     }
+    if (remaining != 0)
```

Comment:
Is this ever going to be true? You have a loop with `remaining > 0` that can be broken only with an exception.